Added event for Meshery design import fail

Yashsharma1911 opened the pull request:
**Notes for Reviewers**

This PR fixes #9259 

This PR adds a new error event on failing Meshery design import to make sure it gets persisted in the notification center

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.
 

<!--
Thank you for contributing to Meshery! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

leecalcote commented on server/handlers/error.go:
```diff
@@ -200,7 +200,7 @@ func ErrPatternFile(err error) error {
 
 func ErrInvalidPattern(err error) error {
-	return errors.New(ErrInvalidPatternCode, errors.Alert, []string{"Invalid design, execution is infeasible", err.Error()}, []string{err.Error()}, []string{"Trait Definition is invalid or unable to process"}, []string{"Check Trait Definition"})
+	return errors.New(ErrInvalidPatternCode, errors.Alert, []string{"Invalid design, execution is infeasible", err.Error()}, []string{err.Error()}, []string{"Services/Traits definition is invalid or unable to process"}, []string{"Ensure that Services/Traits definition present in meshery design", "Ensure that Services/Traits definition is not invalid"})
```
@Yashsharma1911 thanks. We don't use Traits any longer and will hopefully get rid of Services, too. @MUzairS15

Yashsharma1911 replied:
Removing traits from suggestion

leecalcote replied:
This has resurfaced recently, @MUzairS15

MUzairS15 commented on server/handlers/meshery_pattern_handler.go:
```diff
@@ -315,6 +315,11 @@ func (h *Handler) handlePatternPOST(
 				addMeshkitErr(&res, ErrInvalidPattern(err))
+				event := eventBuilder.WithSeverity(events.Error).WithMetadata(map[string]interface{}{
+					"error": ErrInvalidPattern(err),
+				}).WithDescription("invalid design file format").Build()
```
```suggestion
				}).WithDescription("Invalid design file format").Build()
```

MUzairS15 commented on server/handlers/error.go:
```diff
@@ -200,7 +200,7 @@ func ErrPatternFile(err error) error {
 
 func ErrInvalidPattern(err error) error {
-	return errors.New(ErrInvalidPatternCode, errors.Alert, []string{"Invalid design, execution is infeasible", err.Error()}, []string{err.Error()}, []string{"Trait Definition is invalid or unable to process"}, []string{"Check Trait Definition"})
+	return errors.New(ErrInvalidPatternCode, errors.Alert, []string{"Invalid design, execution is infeasible", err.Error()}, []string{err.Error()}, []string{"Services/Traits definition is invalid or unable to process"}, []string{"Ensure that Services/Traits definition present in meshery design", "Ensure that Services/Traits definition is not invalid"})
```
Instead of execution is infeasible, please add the operation name itself. Import/Update/…

leecalcote replied:
@Yashsharma1911 ....

Yashsharma1911 replied:
Make sense updating

leecalcote replied:
Always capitalize Meshery, @Yashsharma1911

leecalcote commented:
Syncing with latest on master.

MUzairS15 commented:
@Yashsharma1911 Were are we heading on this?

leecalcote commented:
@Yashsharma1911 ping, ping.

leecalcote commented:
puh-ping

Yashsharma1911 commented:
I accidentally unsubscribed to this PR 😬 was not getting notifications, I'll pushing the requested changes, thanks for calling on this and giving time

Yashsharma1911 commented:
Updated the error message, I also added this link https://docs.meshery.io/concepts/designs for people to understand design better however we are not doing a good job on that page, I'll update the doc there

Yashsharma1911 commented:
Created this issue to track https://github.com/meshery/meshery/issues/9751

MUzairS15 commented on server/handlers/error.go:
```diff
@@ -187,7 +187,7 @@ func ErrPatternFile(err error) error {
 
 func ErrInvalidPattern(err error) error {
-	return errors.New(ErrInvalidPatternCode, errors.Alert, []string{"Invalid design, execution is infeasible", err.Error()}, []string{err.Error()}, []string{"Trait Definition is invalid or unable to process"}, []string{"Check Trait Definition"})
+	return errors.New(ErrInvalidPatternCode, errors.Alert, []string{"Invalid design, unable to import design", err.Error()}, []string{err.Error()}, []string{"The services definition in your Meshery design is either missing or invalid"}, []string{"Please review your Meshery design to ensure that it includes valid services definitions.", "If you are unfamiliar with Meshery design, consider referring t ...
```
@Yashsharma1911 The `probable cause` and `suggested remediation` should be in pairs i.e. for each cause there is a corresponding remedy. In the UI as well we display as 1-1 mapping b/w these two.

leecalcote replied:
@Althaf66 @RipulHandoo it's good for both of you to be aware of this formatting.

MUzairS15 commented on server/handlers/meshery_pattern_handler.go:
```diff
@@ -313,6 +313,11 @@ func (h *Handler) handlePatternPOST(
 				addMeshkitErr(&res, ErrInvalidPattern(err))
+				event := eventBuilder.WithSeverity(events.Error).WithMetadata(map[string]interface{}{
+					"error": ErrInvalidPattern(err),
+				}).WithDescription("Invalid design file format").Build()
```
Please publish the event as well.

leecalcote commented:
Thanks for reviewing this, @MUzairS15. Some feedback for @Yashsharma1911

leecalcote commented:
@Yashsharma1911 merge conflict.

leecalcote commented:
@Althaf66 perhaps you could assist with this merge conflict.

RipulHandoo commented:
> @Althaf66 perhaps you could assist with this merge conflict.

@leecalcote merge conflict resolved. ✅

leecalcote commented:
Thank you, @RipulHandoo

leecalcote commented:
@MUzairS15, @RipulHandoo might need additional privileges in order to push the commit for eliminated merge conflict.

leecalcote commented:
Syncing branch with `master`.

leecalcote commented:
@Yashsharma1911 all good to go?
